core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -667,6 +678,12 @@ public NativeImageInvokerInfo build() {
                     nativeImageArgs.add("-H:-ParseOnce");
                 }
 
+                if (!nativeImageSecurityProviders.isEmpty()) {
```

Comment:
Since the flag does not exist in GraalVM CE <= 21.1
```suggestion
                if (graalVMVersion.isNewerThan(GraalVM.Version.VERSION_21_1) && !nativeImageSecurityProviders.isEmpty()) {
```

Reply:
You could even nest it inside the previous `if` statement.

Reply:
This line causes an NPE.